### ISSUE TITLE
EPERM lchown error with the put method on v12 and +

### ISSUE THREAD

**Issue by gevorggalstyan** [npm/cacache#3]
After updating to the v12 and + (the current version I use is 12.0.2) I get the following error on put operations:

{ [Error: EPERM: operation not permitted, lchown '/tmp/content-v2/sha512/ec/9e']
  cause:
   { Error: EPERM: operation not permitted, lchown '/tmp/content-v2/sha512/ec/9e'
     errno: -1,
     code: 'EPERM',
     syscall: 'lchown',
     path: '/tmp/content-v2/sha512/ec/9e' },
  isOperational: true,
  errno: -1,
  code: 'EPERM',
  syscall: 'lchown',
  path: '/tmp/content-v2/sha512/ec/9e' }

And I actually see the files and folders created.
With both versions (11+ and 12+) the owner of the folders is my user.

And here is the code portion I use:
```ts
import { put as putCache } from "cacache/en";

const CACHE_PATH = "/tmp";

export const put = async <T>(key: string, payload: T, ttl: number = 10_000) => {
  try {
    await putCache(
      CACHE_PATH,
      key,
      JSON.stringify({
        payload,
        timestamp: Date.now(),
        ttl,
      }),
    );
    return payload;
  } catch (error) {
    console.log("Put Cache", error);
    return null;
  }
};
```

OS: macOS 10.14 and AWS Lambda
Node Version: 10.16.0 and 12.6.0

**Comment by glavoie** [comment]
Just hit this issue, this is caused by this change: https://github.com/npm/cacache/commit/ac84d1478ac0d71b6e556bcf241b52dd1100f1ec

Context: Happened with a `lerna` update which brings this dependency transitively. We build the app in a Docker container using `/tmp` as the `$HOME` folder for a non-root user.

Hi @isaacs! Calling you as the author of this change. Why do you want to automatically infer and `chown` the folder? This assumes that the code runs with `root` permission which should not happen in most cases. 

I'd rather have the code tell me that it cannot write into the cache folder because of wrong permissions and have the user fix the permissions.

**Comment by isaacs** [comment]
It should probably only be doing the lchown when the uid is 0.

> Why do you want to automatically infer and chown the folder? This assumes that the code runs with root permissions which should not happen in most cases.

This prevents cases where a file in the cache is root-owned inappropriately.  If it fails to write the file in the first place, then that error will be raised, but in this case, it has write capabilities, but _not_ lchown capabilities, and we should be able to determine that ahead of time.  I'll fix it on 12.0.3.

**Comment by glavoie** [comment]
Thanks for the explaination. After digging I finally saw the NPM changelog that explains the bigger issue.

What you describe here sounds very reasonable.
